sys/sys/sockopt.h

```diff
@@ -47,7 +47,17 @@ struct socket;
  * Argument structure for sosetopt et seq.  This is in the KERNEL
  * section because it will never be visible to user code.
  */
-enum sopt_dir { SOPT_GET, SOPT_SET, SOPT_GETCAP, SOPT_SETCAP };
+enum sopt_dir {
+	SOPT_GET,
+	SOPT_SET,
+#if __has_feature(capabilities)
+	SOPT_GETCAP,
+	SOPT_SETCAP,
+#else
+#define	SOPT_GETCAP	SOPT_GET
```

Comment:
I see only `uipc_socket.c` using these in the entire tree. Defining these to their non-CAP variants seems a little dodgy, especially given we now have some places that have to key on `__has_feature(capabilities)` to decide whether to check for them or not as otherwise the do the wrong thing for SOPT_GET/SET. Can we not just kill them for non-CHERI so this is always a compile error?

Reply:
Good point, putting all of them under `__has_feature(capabilities)` isn't too awkward.